### PR TITLE
Address some of the new mypy notes about unchecked annotations.

### PR DESCRIPTION
**Pull request by jmchilton**
## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by nsoranzo** [comment]
I think the change in lib/galaxy/tool_util/data/__init__.py will conflict with https://github.com/galaxyproject/galaxy/pull/14971 , would you mind adding them there?

**Comment by jmchilton** [comment]
Thanks for the catch @nsoranzo - I merged the other PR.

File: lib/galaxy/tool_util/data/__init__.py [hunk 1]
```diff
@@ -102,9 +103,30 @@ def exists(self, path: str) -> bool:
...
+)
+LoadInfoT = Tuple[List[Union[str, Element]], Dict[str, Any]]
+
+
 class ToolDataTable(Dictifiable):
     type_key: str
     data: List[List[str]]
+    empty_field_value: str
+    empty_field_values: Dict[str, str]
+    filenames: Dict[str, FileNameInfoT]
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py:127]: Can you please use the same type annotation for `self.filenames` on line 156 below?

File: lib/galaxy/tool_util/data/__init__.py [hunk 2]
```diff
@@ -102,9 +103,30 @@ def exists(self, path: str) -> bool:
...
 
 
+ErrorListT = List[str]
+FileNameInfoT = TypedDict(
+    "FileNameInfoT",
+    {
+        "found": bool,
+        "filename": str,
+        "from_shed_config": bool,
+        "tool_data_path": str,
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
        "tool_data_path": Optional[StrPath],
```

File: lib/galaxy/tool_util/data/__init__.py [hunk 3]
```diff
@@ -102,9 +103,30 @@ def exists(self, path: str) -> bool:
...
+        "config_element": Optional[Element],
+        "tool_shed_repository": Any,
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
        "tool_shed_repository": Optional[Dict[str, Any]],
```

File: lib/galaxy/tool_util/data/__init__.py [hunk 4]
```diff
@@ -102,9 +103,30 @@ def exists(self, path: str) -> bool:
             return os.path.exists(path)
 
 
+ErrorListT = List[str]
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py:106]: You can also use `ErrorListT` in the signature of the `extend_data_with()` and `parse_file_fields()` methods.

File: lib/galaxy/tool_util/data/__init__.py [hunk 5]
```diff
@@ -102,9 +103,30 @@ def exists(self, path: str) -> bool:
...
+        "found": bool,
+        "filename": str,
+        "from_shed_config": bool,
+        "tool_data_path": str,
+        "config_element": Optional[Element],
+        "tool_shed_repository": Any,
+        "errors": ErrorListT,
+    },
+)
+LoadInfoT = Tuple[List[Union[str, Element]], Dict[str, Any]]
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
LoadInfoT = Tuple[Tuple[Element, StrPath], Dict[str, Any]]
```

File: lib/galaxy/tool_util/data/__init__.py [hunk 6]
```diff
@@ -102,9 +103,30 @@ def exists(self, path: str) -> bool:
...
+
+
 class ToolDataTable(Dictifiable):
     type_key: str
     data: List[List[str]]
+    empty_field_value: str
+    empty_field_values: Dict[str, str]
+    filenames: Dict[str, FileNameInfoT]
+    _load_info: LoadInfoT
+    _merged_load_info: List[Tuple[Type, LoadInfoT]]
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/data/__init__.py]:
```suggestion
    _merged_load_info: List[Tuple[Type[ToolDataTable], LoadInfoT]]
```

and can you please use the same type annotation for `self._merged_load_info` on line 173 below?

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**